### PR TITLE
feat: raise PR claim TTL to 15min and add heartbeat renewal to review.md

### PR DESCRIPTION
**Pull request by dmcaulay**
## Summary
- Raises the default PR claim TTL (`SHIPWRIGHT_TASK_STORE_CLAIM_TTL_MS`) from 5 minutes to 15 minutes (`300_000` → `900_000`ms) in `stale-claim-reaper.ts` and the fallback in `pull-request-service.ts`, keeping the value operator-overridable via the env var.
- Syncs the new default across `docs/configuration.md` and `site/src/data/config-vars.ts`.
- Adds an explicit heartbeat renewal step to `review.md` (new Step 5, item 8) between context-gathering and the review-writing/posting steps, so long reviews don't get their claim reaped mid-review.

Root cause: the stale-claim reaper resets `reviewState` to `pending` for any `phase='review'` claim whose heartbeat exceeds the TTL. No command file ever called the heartbeat endpoint, and real reviews often exceed the old 5-minute default — confirmed cause of PR #1016 receiving 5 duplicate review posts against an unchanged commit over 9 hours.

## Acceptance Criteria

| # | Criterion | Status | Evidence |
|---|-----------|--------|----------|
| 1 | Default TTL raised to 900_000ms in stale-claim-reaper.ts + fallback in pull-request-service.ts, still operator-overridable | MET | `DEFAULT_TTL_MS = 900_000` in stale-claim-reaper.ts; `?? 900_000` fallback in pull-request-service.ts; both still read `SHIPWRIGHT_TASK_STORE_CLAIM_TTL_MS` first |
| 2 | docs/configuration.md and config-vars.ts reflect 900000, all three sources in sync | MET | Both docs updated to `900000`, matching the reaper constant |
| 3 | review.md heartbeat step between Step 5 and review-writing/posting | MET | New step inserted after Step 5's context-gathering, before Step 6 (Classify Changes) and Step 7 (Deep Review) |
| 4 | stale-claim-reaper.unit.test.ts extended with 900_000ms boundary case, additive only | MET | 2 new tests added (just-under / just-over TTL boundary), no existing tests retired |

## Test Plan
- `bun test` in `task-store/` — 207 pass, 0 fail (16/16 in `stale-claim-reaper.unit.test.ts`, including 2 new boundary tests)
- `bun run typecheck` in `task-store/` — clean
- [x] Pre-commit checks passing

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: task-store/src/stale-claim-reaper.unit.test.ts [hunk 1]
```diff
@@ -275,4 +275,55 @@ describe("StaleClaimReaper", () => {
     expect(prSql).toContain("'review'");
     expect(prSql).toContain("'pending'");
   });
+
+  // ─── 900_000ms default TTL boundary ────────────────────────────────────────
+
+  test("phase='review' claim just under 900_000ms old is NOT reaped", async () => {
```
Comment (dodizzle) [inline task-store/src/stale-claim-reaper.unit.test.ts:281]: Minor: the comment "simulates a claim just inside the TTL window" slightly overstates what's proven — `affectedRows: 0` is injected by the Prisma double, not filtered by the actual SQL predicate against a real row. The test genuinely validates the cutoff constant and the date arithmetic (the assertions below are correct and sufficient). A one-line clarification like "verifies the cutoff constant and date math" would be more precise. Not blocking.

**Review by dodizzle — APPROVED** [review]
APPROVE — All 4 acceptance criteria met. TTL correctly raised to 900_000ms in both `stale-claim-reaper.ts` and the fallback in `pull-request-service.ts`, docs in sync, heartbeat step placed correctly in `review.md` between the unresolved-comment check and domain classification. Boundary tests validate the cutoff constant and date arithmetic against the new default. CI green, no breaking changes.

**Review by dodizzle — APPROVED** [review]
APPROVE — Minimal, targeted fix for the confirmed #1016 duplicate-review root cause. Both TTL sources (`stale-claim-reaper.ts` constant and `pull-request-service.ts` fallback) raised to 900_000ms identically. Heartbeat renewal step placed correctly in `review.md` (after context-gathering, before subagent dispatch). Docs in sync across 4 files. Boundary tests validate cutoff constant and date math. CI green. All 4 acceptance criteria met. The only change since the prior APPROVE review (`ebb76a91`) is a single test comment improvement.

**Review by dodizzle — APPROVED** [review]
APPROVE (re-review) — Prior suggestion addressed: test comment now reads "verifies the cutoff constant and date math", accurately reflecting what the unit test proves. No new issues. All 4 acceptance criteria remain met, CI green.
